internal/config/config.go

```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/influxdata/telegraf/plugins/inputs"
 	"github.com/influxdata/telegraf/plugins/outputs"
 
-	"github.com/naoina/toml"
+	"github.com/influxdata/config"
 	"github.com/naoina/toml/ast"
```

Comment:
Nope, I didn't wrap the types that came from `naoina/toml/ast` yet since I was focusing on getting everything else working together with this. It looks like it's just a few types that need to be aliased to push that dependency into `influxdata/config`, so I can probably do that.